src/main/java/teammates/ui/webapi/GetRegkeyValidityAction.java

```diff
@@ -27,19 +29,35 @@ void checkSpecificAccessControl() {
     @Override
     public JsonResult execute() {
         Intent intent = Intent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
-        String regkey = getNonNullRequestParamValue(Const.ParamsNames.REGKEY);
+        String regKey = getNonNullRequestParamValue(Const.ParamsNames.REGKEY);
 
         boolean isValid = false;
         String googleId = null;
 
         if (intent == Intent.STUDENT_SUBMISSION || intent == Intent.STUDENT_RESULT) {
-            StudentAttributes student = logic.getStudentForRegistrationKey(regkey);
+            // Try to get googleId for not migrated user
+            StudentAttributes studentAttributes = logic.getStudentForRegistrationKey(regKey);
+            if (studentAttributes != null) {
+                isValid = true;
+                googleId = studentAttributes.getGoogleId();
+            }
```

Comment:
This condition is not enough to determine if the student is migrated. If it's not null, you still have to check if the associated course is migrated or not.

Consider the case where after migration the student the deleted but the old student still exists in datastore.